assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -605,7 +607,12 @@ const RevenueTrip = ({
     " ",
     trip.startPlace || "",
   ].join("")
-  const [{ ladderDirections }] = useContext(StateDispatchContext)
+  const [{ routeTabs }] = useContext(StateDispatchContext)
+
+  const currentTab = currentRouteTab(routeTabs)
+  const ladderDirections = currentTab
```

Comment:
**suggestion (non-blocking):** There's a bunch of variables getting set here, and this long function might be easier to follow if there wasn't quite so much logic spread over 3 variables.

I noticed some similar login in `header.tsx`. Could you extract this into a `import { ladderDirectionsFromRouteTabs } from "routeTab.ts"` function?

Reply:
The main pieces of this logic that aren't already part of `currentRouteTab` are a) pulling `routeTabs` from the context and b) handling the case where `routeTabs` is null. Regarding a, calling `useContext` from a utility function in the `routeTabs` module would violate the [rules of hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions). As for b, the types don't actually guarantee that there will be a unique current tab, hence why null is a possible return value, but based on the application logic that should never happen, so I suppose I could simply raise an exception if that happens and remove null as an option from the return type, simplifying things a little bit. I'm not sure that I want to do that as part of this PR, though.